main.go

```diff
@@ -165,6 +167,16 @@ func main() {
 		Addr: cfg.listenAddr,
 	}
 
+	// Implement maximal connection age, this avoids long running remote_write
+	// connections creating an unbalanced share of load
+	if cfg.maxConnectionAge.Seconds() > 0 {
+		_ = level.Debug(logger).Log("msg", "Setup a max connection age for HTTP connections", "duration", cfg.maxConnectionAge)
+		server.IdleTimeout = cfg.maxConnectionAge
+		server.ConnContext = func(ctx context.Context, _ net.Conn) context.Context {
```

Comment:
nice! i didn't know that `http.Server` provides this hook for modifications to the newly created context when a new connection comes in

Reply:
It is a Go 1.13 feature, but it's a super useful thing and had a quite bad bug: https://github.com/golang/go/issues/35750